tests/assert-jobs/reporter/main.go

```diff
@@ -32,13 +56,128 @@ func initTracer() (opentracing.Tracer, io.Closer) {
+// services: number of services to generate
+func generateSpansHistory(serviceName, operationName string, days, services int) {
+	for service := 0; service < services; service++ {
+		reportedServiceName := serviceName
+		if services > 1 {
+			reportedServiceName = fmt.Sprintf("%s-%d", serviceName, service)
+		}
+		generateSpansHistoryService(reportedServiceName, operationName, days)
+	}
+}
+
+// Block the execution until the Jaeger REST API is available (or timeout)
+func waitUntilRestAPIAvailable(jaegerEndpoint string) error {
+	transport := &http.Transport{}
+	client := http.Client{Transport: transport}
+
+	err := wait.Poll(time.Second*5, time.Minute*5, func() (done bool, err error) {
+		req, err := http.NewRequest(http.MethodGet, jaegerEndpoint, nil)
+		if err != nil {
+			return false, err
+		}
+
+		resp, err := client.Do(req)
+
+		// The GET HTTP verb is not supported by the Jaeger Collector REST API enpoint. An error 405
+		// means the REST API is there
+		if resp != nil && resp.StatusCode == 405 {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+
+		log.Warningln(jaegerEndpoint, "is not available. Is", envVarJaegerEndpoint, "environment variable properly set?")
+		return false, nil
+	})
+	return err
+}
+
+// Init the CMD and return error if something didn't go properly
+func initCmd() error {
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	viper.AutomaticEnv()
+
+	viper.SetDefault(flagJaegerServiceName, "jaeger-service")
+	flag.String(flagJaegerServiceName, "", "Jaeger service name")
+
+	viper.SetDefault(flagDays, 1)
+	flag.Int(flagDays, 1, "History days")
+
+	viper.SetDefault(flagServices, 1)
+	flag.Int(flagServices, 1, "Number of services")
+
+	viper.SetDefault(flagVerbose, false)
+	flag.Bool(flagVerbose, false, "Enable verbosity")
+
+	viper.SetDefault(flagJaegerOperationName, "jaeger-operation")
+	flag.String(flagJaegerOperationName, "", "Jaeger operation name")
+
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.Parse()
+
+	err := viper.BindPFlags(pflag.CommandLine)
+	return err
+}
+
+func main() {
+	err := initCmd()
+	if err != nil {
+		log.Error(err)
+		os.Exit(1)
+	}
+
+	log = *logger.InitLog(viper.GetBool(flagVerbose))
+
+	jaegerEndpoint := viper.GetString(envVarJaegerEndpoint)
+
+	if jaegerEndpoint == "" {
+		log.Fatalln("Please, specify a Jaeger endpoint")
+		os.Exit(1)
+	}
+
+	// Sometimes, Kubernetes reports the Jaeger service is there but there is an interval where the service is up but the
+	// REST API is not operative yet
+	err = waitUntilRestAPIAvailable(jaegerEndpoint)
+	if err != nil {
+		log.Fatalln(err)
+		os.Exit(1)
+	}
+
+	generateSpansHistory(viper.GetString(flagJaegerServiceName), viper.GetString(flagJaegerOperationName), viper.GetInt(flagDays), viper.GetInt(flagServices))
+
+	// After reporting the spans, we wait some seconds to ensure the spans were reported and
+	// stored in the final storage (ES or other)
+	time.Sleep(time.Second * 10)
```

Comment:
Can we put this in a constant or parameter "`waiting_after_reporting`" or something like that?